app/stores/data-source.ts

```diff
@@ -17,7 +17,13 @@ type DataSourceStore = {
 
 const STORAGE_KEY = "dataSource";
 
-const customPersist =
+/**
+ * Custom middleware that saves data source to localStorage.
+ *
+ * On initialization it tries to first retrieve data source from URL; if it's
+ * not here, tries with localStorage, otherwise uses a default data source.
+ */
```

Comment:
Would be helpful to know how the value is stored in the router and in localStorage.